### ISSUE TITLE
Send Error Code 400, but the information has been sent...

### ISSUE THREAD

**Issue by prickles** [Esri/Survey123Community#207]
I'm new to Survey123 and have created a survey with the following types of fields: location, select one, text, and image (for users to add a photo). The survey's loaded fine and I'm able to complete an observation; however, when I go to send the information, it says Send Error Code 400 stating that it is unable to send the information (and displays a little blue cloud with a red exclamation point). I tried to resend two more times, but received the same error (and the blue cloud with the red exclamation point is still there); however, when I checked the Web Map in ArcGIS Online, I can see my point and two copies of it (so all the times the observation were sent, they were successful). I've tried this on a Moto G as well as an iPhone and has occurred on both; are there any suggestions for fixing the issue?

**Comment by prickles** [comment]
Update: It add the point to the map but NOT the image, so that's probably why it's throwing an issue. Any guidance?

**Comment by survey123** [comment]
Hello @prickles! 

When Survey123 is adding records to your data, it will add the features first and then any attachments (e.g., photos). It looks as though your internet connection is not reliable enough to complete the entire upload at once (e.g., the features are added, but the connection is lost before the photo could be uploaded).

Is it possible to try again on your devices using a stronger, more reliable wifi signal?

**Comment by prickles** [comment]
Thanks for the quick response; I can confirm that I've tried to sync across 3 devices (one android smartphone, one iPhone, and one iPad) trying two separate wifi connections and 3g and am receiving the same issue on all of them (points will be transferred but images are not, and there's a blue cloud with a red exclamation point). Is there anything else we can troubleshoot?

**Comment by bcgisjosh** [comment]
i'm getting the same issue
error 400 for my colleague
I'm getting Code 1000 - No maping exists from object type ESRI.ArcGIS.Client.Geometry.MapPoint to a known managed provider native type.

**Comment by survey123** [comment]
@prickles  @bcgisjosh 
We believe this may have to do with the size of the photos.
We will look into this for a fix.
Are you pulling the photos from the library, or are you submitting a new photo that you took from the form?

**Comment by prickles** [comment]
@survey123 trying to take a new photo

**Comment by thakman** [comment]
I get error 400 when I attach a picture from the device camera roll. If I attach a picture directly with the camera, it sends without error.

**Comment by gkburgo** [comment]
I'd like to add myself to the list of Code 400 send errors. My photo was from my library. Is there a way to delete an attachment from the unsent form?

**Comment by TViveiros** [comment]
I'm new to Survey 123 and am getting a similar error when attempting to submit a survey.  The error reads: Send Error. This survey was not able to be sent due to the following error  Code 1000: No mapping exists from object type Esri ArcGIS Client Geometry MapPoint to a known managed provider native type.  I am using Version 1.2.18 of Survey123ConnectBeta and Version 1.2.77 of Survey 123.  My survey does not include any attachments.

**Comment by HannahFerrier** [comment]
@TViveiros Your error 1000 is caused by binding data into an esri field type that it is unable to support. Please look at all of the questions that you have bound to esri field types and remove incompatible types to narrow down this issue. Your data is able to be successfully captured in the form, but the field type in the feature service is unable to accept the data. Let me know how you get on :)

**Comment by jeff3780** [comment]
I can also confirm that selecting my image from the camera roll, as opposed to capturing using the camera itself is resulting in a Code 400 error when I submit my form.  The point feature is created, and all form attributes are populated, but no attachment is sent to the feature service.  I tried this with an image field in a main form, and then separately in a form where I have the image field in a repeats block.  In both situations, as long as my image comes from the camera, it uploads with the form just fine,  In the repeats block, all my related records get their image if it came from the camera.  I think you know about this issue and are working on it.  Something to do with file sizes.  I'll look forward to a resolution as I noticed the camera control is only able to use the forward camera.  If I wanted to capture an image with the rear facing camera, my thinking as a workaround is to take the picture with the device camera flipped around to rear facing, then invoke my form and attach the image from the camera roll.  Since images from the camera roll are not working right now, I guess there would be no way to add an image to the form from the rear camera.  Most smart phones nowadays have the two cameras available.  I'm not sure how much need there would be to capture an image from either camera, but if the camera roll support is fixed, then at least there would be a workaround that could support images from either camera.
